presto-jdbc/src/main/java/com/facebook/presto/jdbc/AbstractConnectionProperty.java

```diff
@@ -192,6 +196,29 @@ private StringMapConverter() {}
         }
     }
 
+    protected static final class CustomHeadersConverter
+            implements Converter<Map<String, String>>
+    {
+        private static final CharMatcher PRINTABLE_ASCII = CharMatcher.inRange((char) 0x21, (char) 0x7E);
+        public static final CustomHeadersConverter CUSTOM_HEADERS_CONVERTER = new CustomHeadersConverter();
+        ObjectMapper objectMapper = new ObjectMapper();
+        private CustomHeadersConverter()
+        {
+            objectMapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+        }
+
+        @Override
+        public Map<String, String> convert(String value)
+        {
+            try {
+                return objectMapper.readValue(value, HashMap.class);
```

Comment:
to get rid of the Unchecked assignment compiler warning
use  `TypeReference` instead.
`return objectMapper.readValue(value, new TypeReference<HashMap<String, String>>() {});`

Reply:
✅